src/App.js

```diff
@@ -1,5 +1,55 @@
+const { Console } = require("@woowacourse/mission-utils");
+const { MESSAGE, GAME_PROGRESS, ANSWER_LENGTH } = require("./constants/constants");
+const countBallAndStrike = require("./utils/countBallAndStrike");
+const makeRandomNumber = require("./utils/makeRandomNumber");
+const printGameMessage = require("./utils/printGameMessage");
+const verifyInputNumber = require("./utils/verifyInputNumber");
+
 class App {
-  play() {}
+  play() {
+    Console.print(MESSAGE.START_GAME);
+    this.gamePrepare();
+  }
+
+  // 게임 준비 (정답 준비)
+  gamePrepare() {
+    const randomNum = makeRandomNumber();
+    this.gameStart(randomNum);
+  }
+
+  // 게임 시작
+  gameStart(answer) {
+    Console.readLine(MESSAGE.INPUT_NUMBER, (userNumber) => {
+      // 사용자 입력값 유효 여부 확인
+      if (!verifyInputNumber(userNumber)) {
+        this.inputErrorException();
+      }
+      const userResult = countBallAndStrike(userNumber, answer); // 볼 스트라이크 개수 세기
+      printGameMessage(userResult.strike, userResult.ball); // 볼 스트라이크 출력
+      if (userResult.strike < ANSWER_LENGTH) {
+        return this.gameStart(answer);
+      }
+      this.gameRestartCheck();
+    });
+  }
+
```

Comment:
callback을 인라인으로선언하지 않고 외부에서 함수를 정의해서 넘겨주는 방법을 고민해보시면 좋을것같아요!:)

Reply:
@nerdchanii 조언 감사합니다! 궁금한 점이 있는데,
callback에 모듈로 분리해뒀던 verifyInputNumber, countBallAndStrike, printGameMessage가 모두 들어가서 이 부분을 분리했거든요! 
이 기능을 다시 App.js로 호출하게 되면 기능간 의존성들이 너무 강해질 것 같아서 이대로 작성하였는데, 
App.js -> 콜백부분 -> verifyInputNumber 과 같이 두 다리정도는 괜찮으려나요??

Reply:
콜백을 트리거 정도로 보시면 어떨까요?!
구현체 없이 코드베이스를 생각해보면 이럴것 같아요!
```suggestion
  play(){
    app.gamStart(makeRandomNumber())
    }
    
    gamStart(answer){
   constole.readline(message, (input)=>ingameRoutine(answer, input))
   }
   
   ingameRoutine(answer, userInput){
     if (!verifyInputNumber(userNumber)) {
        this.inputErrorException();
      }
      const userResult = countBallAndStrike(userNumber, answer); // 볼 스트라이크 개수 세기
      printGameMessage(userResult.strike, userResult.ball); // 볼 스트라이크 출력
      if (userResult.strike < ANSWER_LENGTH) {
        return this.gameStart(answer);
      }
      this.gameRestartCheck()
   }
```

Reply:
아하 이런 방식으로 말씀이시군요!
ingameRoutine으로 분해해서 이렇게 만들어서 하면 생각했던것보다는 훨씬 결합성이 약해지네요! gameStart()에 추가적인 로직이 더 들어간다면 훨씬 코드읽기도 깔끔해질 것 같아서, 좋은 방법인 것 같아요!
감사합니다ㅎㅎ